src/server/scripts/EasternKingdoms/Stratholme/instance_stratholme.cpp

```diff
@@ -474,91 +573,118 @@ class instance_stratholme : public InstanceMapScript
+                    instance->SummonCreature(NPC_MINDLESS_UNDEAD, pos);
+                    break;
+                }
                 case EVENT_FORCE_SLAUGHTER_EVENT:
+                {
+                    Map::PlayerList const& PlayerList = instance->GetPlayers();
+                    if (!PlayerList.IsEmpty())
+                        for (Map::PlayerList::const_iterator i = PlayerList.begin(); i != PlayerList.end(); ++i)
+                            if (Player* player = i->GetSource())
+                                if (player->GetDistance2d(4034.97f, -3402.13f) < 50.0f)
+                                {
+                                    ProcessSlaughterEvent();
+                                    return;
+                                }
+
+                    events.ScheduleEvent(EVENT_FORCE_SLAUGHTER_EVENT, 3s);
+                    break;
+                }
+                case EVENT_SPAWN_BLACK_GUARD:
+                {
+                    for (uint8 i = 0; i < 5; ++i)
+                        if (Creature* guard = instance->SummonCreature(NPC_BLACK_GUARD, BlackGuardPos[i]))
+                        {
+                            guard->SetWalk(true);
+                            guard->GetMotionMaster()->MovePoint(0, BlackGuardPos[i + 5]);
+                            guard->SetHomePosition(BlackGuardPos[i + 5]);
+                            if (i == 0 && guard->AI())
+                                guard->AI()->Talk(SAY_BLACK_GUARD_INIT);
+                        }
+
+                    if (GameObject* gate = instance->GetGameObject(_zigguratDoorsGUID4))
+                        gate->SetGoState(GO_STATE_ACTIVE);
+                    break;
+                }
+                case EVENT_BARTHILAS_MOVE_END:
+                    if (Creature* barthilas = instance->GetCreature(_barthilasGUID))
                     {
-                        Map::PlayerList const& PlayerList = instance->GetPlayers();
-                        if (!PlayerList.IsEmpty())
-                            for (Map::PlayerList::const_iterator i = PlayerList.begin(); i != PlayerList.end(); ++i)
-                                if (Player* player = i->GetSource())
-                                    if (player->GetDistance2d(4034.97f, -3402.13f) < 50.0f)
-                                    {
-                                        ProcessSlaughterEvent();
-                                        return;
-                                    }
-
-                        events.ScheduleEvent(EVENT_FORCE_SLAUGHTER_EVENT, 3s);
-                        break;
+                        if (barthilas->isDead())
+                            break;
+                        if (barthilas->IsAlive() && (barthilas->GetPosition()) == Position(4068.28f, -3535.67f, 122.771f, 2.50f))
+                        {
+                            barthilas->SetHomePosition(4068.28f, -3535.67f, 122.771f, 2.50f);
+                            SetData(TYPE_BARTHILAS_RUN, DONE);
```

Comment:
Based on sniffed values. Still needs to be moved to the top.

```suggestion
                        if (barthilas->IsAlive() && (barthilas->GetPosition()) == Position(4068.74f, -3535.97f, 122.825f, 0.2521562f))
                        {
                            barthilas->SetHomePosition(4068.74f, -3535.97f, 122.825f, 0.2521562f);
                            SetData(TYPE_BARTHILAS_RUN, DONE);
```